pkg/models/vulnerability.go

```diff
@@ -2,17 +2,24 @@ package models
 
 import (
 	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
 	"time"
+
+	"github.com/google/osv-scanner/internal/semantic"
+	"github.com/google/osv-scanner/pkg/lockfile"
 )
 
 // Package identifies the affected code library or command provided by the
 // package.
 //
 // See: https://ossf.github.io/osv-schema/#affectedpackage-field
 type Package struct {
-	Ecosystem Ecosystem `json:"ecosystem" yaml:"ecosystem"`
-	Name      string    `json:"name" yaml:"name"`
-	Purl      string    `json:"purl,omitempty" yaml:"purl,omitempty"`
+	Ecosystem lockfile.Ecosystem `json:"ecosystem" yaml:"ecosystem"`
```

Comment:
This is kind of a cop-out to get things working - I think technically this is a breaking change, but ultimately I think a decision needs to be made on how these constants are handled since now defined in at least two places and I'm pretty sure reconciling that would require a breaking change too technically?

(I'm pretty sure alternatively I can force Go to convert whenever these two types meet to avoid the breaking change, but have gone with the easier route for now until we've discussed since I could be wrong 🤷)

Reply:
Let's avoid breakingthe type here, as we have a few consumers of this package here. Converting between them is a little annoying, but still easily doable. 

It's also nice to reduce the dependencies between the `lockfile` and the `models` packages.